WalletWasabi.Gui/Tabs/SettingsViewModel.cs

```diff
@@ -15,395 +15,463 @@
-					var strongPrivacyLevel = int.TryParse(StrongPrivacyLevel, out level) ? (int?)level : null;
-					var dustThreshold = decimal.TryParse(DustThreshold, out var threshold) ? (decimal?)threshold : null;
-
-					var (configLocalHost, configLocalPort) = network == NBitcoin.Network.Main
-						? (config.MainNetBitcoinCoreHost, config.MainNetBitcoinCorePort)
-						: (network == NBitcoin.Network.TestNet
-							? (config.TestNetBitcoinCoreHost, config.TestNetBitcoinCorePort)
-							: (config.RegTestBitcoinCoreHost, config.RegTestBitcoinCorePort));
-
-					if (config.Network != network
-						|| config.TorHost != torHost
-						|| config.TorSocks5Port != torSocks5Port
-						|| config.UseTor != useTor
-						|| config.PrivacyLevelSome != somePrivacyLevel
-						|| config.PrivacyLevelFine != finePrivacyLevel
-						|| config.PrivacyLevelStrong != strongPrivacyLevel
-						|| config.DustThreshold.ToUnit(MoneyUnit.BTC) != dustThreshold
-						|| configLocalHost != localNodeHost
-						|| configLocalPort.ToString() != localNodePort
-					)
-					{
-						config.Network = network;
-						config.TorHost = torHost;
-						config.TorSocks5Port = torSocks5Port;
-						config.UseTor = useTor;
-						config.PrivacyLevelSome = somePrivacyLevel;
-						config.PrivacyLevelFine = finePrivacyLevel;
-						config.PrivacyLevelStrong = strongPrivacyLevel;
-						config.DustThreshold = Money.Coins(dustThreshold.Value);
-
-						switch (network.Name)
-						{
-							case "Main":
-								config.MainNetBitcoinCoreHost = localNodeHost;
-								config.MainNetBitcoinCorePort = int.Parse(localNodePort);
-								break;
-
-							case "TestNet":
-								config.TestNetBitcoinCoreHost = localNodeHost;
-								config.TestNetBitcoinCorePort = int.Parse(localNodePort);
-								break;
-
-							case "RegTest":
-								config.RegTestBitcoinCoreHost = localNodeHost;
-								config.RegTestBitcoinCorePort = int.Parse(localNodePort);
-								break;
-						}
-
-						await config.ToFileAsync();
-					}
-
-					IsModified = await Global.Config.CheckFileChangeAsync();
-				}
-			});
-		}
-
-		public string ValidateTorHost()
-			=> ValidateHost(TorHost);
-
-		public string ValidateLocalNodeHost()
-			=> ValidateHost(LocalNodeHost);
-
-		public string ValidateTorPort()
-			=> ValidatePort(TorPort);
-
-		public string ValidateLocalNodePort()
-			=> ValidatePort(LocalNodePort);
-
-		public string ValidateSomePrivacyLevel()
-			=> ValidatePrivacyLevel(SomePrivacyLevel);
-
-		public string ValidateFinePrivacyLevel()
-			=> ValidatePrivacyLevel(FinePrivacyLevel);
-
-		public string ValidateStrongPrivacyLevel()
-			=> ValidatePrivacyLevel(StrongPrivacyLevel);
-
-		public string ValidatePrivacyLevel(string value)
-		{
-			if (string.IsNullOrWhiteSpace(value))
-			{
-				return string.Empty;
-			}
-
-			if (uint.TryParse(value, out _))
+    internal class SettingsViewModel : WasabiDocumentTabViewModel
```

Comment:
Hmm.. Auto-formatting on VSCode isnt respecting the editorconfig indent policy at all..